### PR TITLE
[MIG] account_operating_unit: Migration to 17.0

### PR DESCRIPTION
**Pull request by ghost**
Depends on:

- [x] https://github.com/OCA/operating-unit/pull/667
- [x] https://github.com/OCA/operating-unit/pull/669

### CODE REVIEW

**Comment by jdidderen-nsi** [comment]
As soon as #667 is merged, everything should be good to go for this one.

File: account_operating_unit/models/account_move.py
Comment (yankinmax) [inline account_operating_unit/models/account_move.py]: why is is just removed?

File: account_operating_unit/models/account_move.py
Comment (yankinmax) [inline account_operating_unit/models/account_move.py]: you don't need to explicitly assign an `id` to a many2one record, just `record.journal_id.operating_unit_id`

File: account_operating_unit/report/account_invoice_report.py
Comment (yankinmax) [inline account_operating_unit/report/account_invoice_report.py]: why is it just removed?

File: account_operating_unit/views/account_move_view.xml
Comment (yankinmax) [inline account_operating_unit/views/account_move_view.xml:85]: `operation="update"` will be only applied if you have a dependency on `base_view_inheritance_extension` I suppose, so probably you want to overwrite the original context like it was before

**Comment by yankinmax** [comment]
> As soon as #667 and #669 is merged, everything should be good to go for this one.

why don't simply add `tets-requirements.txt` file including dependencies references?
just name that commit with prefix `DON'T MERGE`

File: account_operating_unit/report/account_invoice_report.py
Comment (ghost) [reply account_operating_unit/report/account_invoice_report.py]: It was removed from OC in v16 (https://github.com/odoo/odoo/pull/96134) and it seems that it's not used anymore outside of reports like account.invoice.report

File: account_operating_unit/models/account_move.py
Comment (ghost) [reply account_operating_unit/models/account_move.py]: It was removed from OC in v16 (https://github.com/odoo/odoo/pull/96134) and it seems that it's not used anymore outside of reports like account.invoice.report

File: account_operating_unit/views/account_move_view.xml
Comment (ghost) [reply account_operating_unit/views/account_move_view.xml:85]: Yes it's exactly that

**Comment by ghost** [comment]
> > As soon as #667 and #669 is merged, everything should be good to go for this one.
> 
> why don't simply add `tets-requirements.txt` file including dependencies references? just name that commit with prefix `DON'T MERGE`

Thanks for the tip :)

File: account_operating_unit/models/account_move.py
Comment (ghost) [reply account_operating_unit/models/account_move.py]: Updated. Thanks for the review 👍

**Review by yankinmax — APPROVED** [review]

File: account_operating_unit/tests/test_payment_operating_unit.py
Comment (Camille0907) [inline account_operating_unit/tests/test_payment_operating_unit.py:79]: No risk of float precision issue ?

File: account_operating_unit/models/account_move.py
Comment (Camille0907) [inline account_operating_unit/models/account_move.py:218]: Is `check_move_validity` a new key you just created or already existing somewhere ?
Asking because if it's a new one, it isn't read anywhere, so I don't understand its purpose

**Review by Camille0907 — APPROVED** [review]
Migration technically LGTM

File: account_operating_unit/models/account_move.py
Comment (ghost) [reply account_operating_unit/models/account_move.py:218]: It's a key added in Odoo Community a few version ago (https://github.com/odoo/odoo/blob/96223d322c0f59b35ccda01c82e99a1ac6e2fe0c/addons/account/models/account_move.py#L1890). 
The purpose is to avoid the check on the balance.

File: account_operating_unit/tests/test_payment_operating_unit.py
Comment (ghost) [reply account_operating_unit/tests/test_payment_operating_unit.py:79]: I don't think so. Since the decimal precision of the amount field is based on the currency, we should not have any difference between the amount of the invoice and the amount of the payment.

File: account_operating_unit/models/account_move.py
Comment (Camille0907) [reply account_operating_unit/models/account_move.py:218]: Alright; thank you :)

**Comment by yankinmax** [comment]
@jdidderen-nsi can you please remove dependency commit and rebase?
operating_unit is merged:
- [x] https://github.com/OCA/operating-unit/pull/667

**Comment by ghost** [comment]
@yankinmax And it's rebased 👍

**Comment by hbrunn** [comment]
I was curious how you solved adding the operating unit to the analytic distribution widget, as I mentioned in the [v16 migration](https://github.com/OCA/operating-unit/pull/656#issue-2186979149), but it seems you didn't, correct? In my opinion this needs to be done for the migration to be complete

**Comment by ghost** [comment]
@hbrunn It was solved directly in the analytic_operating_unit module - https://github.com/OCA/operating-unit/pull/669

**Comment by hbrunn** [comment]
ah nice, thanks. That would then also be a good solution for v16, will backport that eventually

**Review by hbrunn — APPROVED** [review]
please also cherry pick https://github.com/OCA/operating-unit/pull/685 so that we have feature parity

**Comment by hbrunn** [comment]
please also add https://github.com/OCA/operating-unit/pull/656/commits/d328870c717f758f1af0eab7e3da6de118c9d602 for higher test coverage

**Comment by yankinmax** [comment]
@jdidderen-nsi can you pls apply @hbrunn last suggestions?

**Comment by ghost** [comment]
> please also cherry pick #685 so that we have feature parity

> please also add [d328870](https://github.com/OCA/operating-unit/commit/d328870c717f758f1af0eab7e3da6de118c9d602) for higher test coverage

> @jdidderen-nsi can you pls apply @hbrunn last suggestions?

@hbrunn @yankinmax And it's done 👍

**Comment by AaronHForgeFlow** [comment]
Issue with pre-commit on build, trying it again

/ocabot merge nobump

**Review by AaronHForgeFlow — APPROVED** [review]

**Pull request merged**
